### DISCUSSION TITLE
Get redis database number from RedisDsn

### DISCUSSION THREAD

**Discussion by Trinkes** [pydantic/pydantic#2820] [Question]
Hello,

When loading redis url to my sistem, i'm using pydantic BaseSettings mechanism with a `RedisDsn` field type and setting as environment variable an url like `redis://127.0.0.1:6379/0`.
When instantiating the redis class, is there any way of providing the hole RedisDsn variable or some method to facilitate the instantiation?

Currently i have the following:
```python
redis: Redis = Redis(
            host=settings.REDIS_URL.host,
            port=settings.REDIS_URL.port,
            db=settings.REDIS_URL.path.replace("/", ""),
            password=settings.REDIS_URL.password,
            username=settings.REDIS_URL.user,
        )
```

Other question, is there anyway of getting db number without having to parse it by myself? A `settings.REDIS_URL.db_number` would be perfect.

Thank you in advance.

**Comment by nuno-andre** [comment]
Something like a [`slug` property](https://en.wikipedia.org/wiki/Clean_URL#Slug) in `AnyUrl` would be a perfect fit for this use case (`settings.REDIS_URL.slug` would be `0` in this example), and many others, e.g. restful resources.
